src/qrules/particle.py

```diff
@@ -101,16 +114,23 @@ def __repr__(self) -> str:
 
     def _repr_pretty_(self, p: PrettyPrinter, _: bool) -> None:
         class_name = type(self).__name__
-        magnitude = _to_fraction(self.magnitude)
-        projection = _to_fraction(self.projection, render_plus=True)
+        magnitude = _to_signed_fraction(self.magnitude)
+        projection = _to_signed_fraction(self.projection, render_plus=True)
         p.text(f"{class_name}({magnitude}, {projection})")
 
 
+def _to_signed_fraction(fraction: Fraction, render_plus: bool = False) -> str:
+    string_representation = str(fraction)
+    if render_plus and fraction.numerator > 0:
+        return f"+{string_representation}"
+    return string_representation
+
+
 def _to_parity(value: Parity | int) -> Parity:
     return Parity(int(value))
 
 
-def _to_spin(value: Spin | tuple[float, float]) -> Spin:
+def _to_spin(value: Spin | tuple[Fraction, Fraction]) -> Spin:
```

Comment:
Maybe this?

```suggestion
def _to_spin(value: Spin | tuple[SupportsFloat, SupportsFloat]) -> Spin:
```

See also elsewhere

Reply:
As in `_halves_domain`, `SupportsFloat` creates problems in mypy with `Fraction`